firedrake/preconditioners/assembled.py

```diff
@@ -44,7 +44,8 @@ def initialize(self, pc):
 
         (a, bcs) = self.form(pc, test, trial)
 
-        form_assembler = get_assembler(a, bcs=bcs, form_compiler_parameters=fcp, mat_type=mat_type, options_prefix=options_prefix)
+        bcs_Jp = tuple(bc.extract_form('Jp') for bc in bcs)
```

Comment:
This should go in `self.form` so every preconditioner gets the correct bcs

Reply:
Thanks.